scripts/add-contribution.js

```diff
@@ -26,23 +27,32 @@ module.exports = async function(callback) {
 
   console.log(`Creating a contribution for contributor account ${contributorAccount} ID: ${contributorId}`);
 
+  [ dateNow, timeNow ] = (new Date()).toISOString().split('T');
+
   let contributionAttributes = {
     contributorId,
+    date: dateNow,
+    time: timeNow,
     amount: await promptly.prompt('Amount: '),
     description: await promptly.prompt('Description: '),
     kind: await promptly.prompt('Kind: ', { default: 'dev' }),
     url: await promptly.prompt('URL: ', { default: '' })
   }
 
+  const contributorData = await kredits.Contributor.getById(contributorId);
+  contributionAttributes.contributorIpfsHash = contributorData.ipfsHash;
+
   console.log("\nAdding contribution:");
   console.log(contributionAttributes);
 
-  kredits.Contribution.addContribution(contributionAttributes, { gasLimit: 300000 }).then((result) => {
-    console.log("\n\nResult:");
-    console.log(result);
-    callback();
-  }).catch((error) => {
-    console.log('Failed to create contribution');
-    callback(error);
-  });
+  kredits.Contribution.addContribution(contributionAttributes, { gasLimit: 300000 })
+    .then(result => {
+      console.log("\n\nResult:");
+      console.log(result);
+      callback();
+    })
+    .catch(error => {
+      console.log('Failed to create contribution');
+      callback(inspect(error));
```

Comment:
does this work? thought the callback should get an error object.

Reply:
We need to reject with the tv4 error object either way. It does show the error and doesn't stall. So it works well enough to be usable, yes.